x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -178,215 +202,365 @@ export class ExecutionHandler<
+        })
+      ) {
+        if (!ruleRunMetricsStore.hasConnectorTypeReachedTheLimit(actionTypeId)) {
           logger.debug(
-            `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions has been reached.`
+            `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions for connector type ${actionTypeId} has been reached.`
           );
-          break;
         }
 
-        if (
-          ruleRunMetricsStore.hasReachedTheExecutableActionsLimitByConnectorType({
-            actionTypeId,
-            actionsConfigMap,
-          })
-        ) {
-          if (!ruleRunMetricsStore.hasConnectorTypeReachedTheLimit(actionTypeId)) {
-            logger.debug(
-              `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions for connector type ${actionTypeId} has been reached.`
-            );
-          }
-          ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
-            actionTypeId,
-            status: ActionsCompletion.PARTIAL,
-          });
-          continue;
+        ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
+          actionTypeId,
+          status: ActionsCompletion.PARTIAL,
+        });
+
+        continue;
+      }
+
+      if (!this.isExecutableAction(action)) {
+        this.logger.warn(
+          `Rule "${this.taskInstance.params.alertId}" skipped scheduling action "${action.id}" because it is disabled`
+        );
+
+        continue;
+      }
+
+      ruleRunMetricsStore.incrementNumberOfTriggeredActions();
+      ruleRunMetricsStore.incrementNumberOfTriggeredActionsByConnectorType(actionTypeId);
+
+      if (summarizedAlerts && !isSystemAction(action)) {
+        const { actionsToEnqueueForExecution, actionsToLog } = await this.runSummarizedAction({
+          action,
+          summarizedAlerts,
+          spaceId,
+        });
+
+        enqueueActions.push(...actionsToEnqueueForExecution);
+        Object.assign(logActions, actionsToLog);
+
+        if (isActionOnInterval(action)) {
+          throttledSummaryActions[action.uuid!] = { date: new Date().toISOString() };
         }
+      } else if (isSystemAction(action) && summarizedAlerts) {
+        const hasConnectorAdapter = this.taskRunnerContext.connectorAdapterRegistry.has(
+          action.actionTypeId
+        );
 
-        if (!this.isExecutableAction(action)) {
+        /**
+         * System actions without an adapter
+         * cannot be executed
+         *
+         */
+        if (!hasConnectorAdapter) {
           this.logger.warn(
-            `Rule "${this.taskInstance.params.alertId}" skipped scheduling action "${action.id}" because it is disabled`
+            `Rule "${this.taskInstance.params.alertId}" skipped scheduling system action "${action.id}" because no connector adapter is configured`
           );
```

Comment:
**question:** should we throw an error instead similar to when a connector doesn't exist?